libs/auth/src/effects/auth.effects.spec.ts

```diff
@@ -0,0 +1,257 @@
+import { Observable ,  of } from 'rxjs';
+import { hot, cold } from 'jasmine-marbles';
+
+import {
+  DaffAccountRegistrationFactory,
+  DaffAuthTokenFactory
+} from '@daffodil/auth/testing';
+
+import { DaffAuthEffects } from './auth.effects';
+import {
+  DaffAuthLogin,
+  DaffAuthLoginSuccess,
+  DaffAuthLoginFailure,
+  DaffAuthRegister,
+  DaffAuthRegisterSuccess,
+  DaffAuthRegisterFailure,
+  DaffAuthCheck,
+  DaffAuthCheckSuccess,
+  DaffAuthCheckFailure,
+  DaffAuthLogout,
+  DaffAuthLogoutSuccess,
+  DaffAuthLogoutFailure
+} from '../actions/auth.actions';
+import { DaffRegisterDriver } from '../drivers/interfaces/register-service.interface';
+import { DaffLoginDriver } from '../drivers/interfaces/login-service.interface';
+import { DaffAccountRegistration } from '../models/account-registration';
+import { DaffAuthToken } from '../models/auth-token';
+import { DaffLoginInfo } from '../models/login-info';
+import { DaffAuthDriver } from '../drivers/interfaces/auth-service.interface';
+
+describe('DaffAuthEffects', () => {
+  let actions$: Observable<any>;
+  let effects: DaffAuthEffects<
+    DaffLoginInfo,
+    DaffAuthToken,
+    DaffAccountRegistration
+  >;
+
+  let daffLoginDriver;
+  let daffRegisterDriver;
+  let daffAuthDriver;
+
+  const registrationFactory: DaffAccountRegistrationFactory = new DaffAccountRegistrationFactory();
+  const authFactory: DaffAuthTokenFactory = new DaffAuthTokenFactory();
+
+  let mockAuth: DaffAuthToken;
+  let mockLoginInfo: DaffLoginInfo;
+  let token: string;
+  let email: string;
+  let password: string;
+  let firstName: string;
+  let lastName: string;
+  let mockRegistration: DaffAccountRegistration;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        DaffAuthEffects,
+        provideMockActions(() => actions$),
+        {
+          provide: DaffLoginDriver,
+          useValue: jasmine.createSpyObj('DaffLoginService', ['login', 'logout'])
+        },
+        {
+          provide: DaffRegisterDriver,
+          useValue: jasmine.createSpyObj('DaffRegisterService', ['register'])
+        },
+        {
+          provide: DaffAuthDriver,
+          useValue: jasmine.createSpyObj('DaffAuthService', ['check'])
+        }
+      ]
+    });
+
+    effects = TestBed.get(DaffAuthEffects);
+
+    daffLoginDriver = TestBed.get(DaffLoginDriver);
+    daffRegisterDriver = TestBed.get(DaffRegisterDriver);
+    daffAuthDriver = TestBed.get(DaffAuthDriver);
+
+    mockRegistration = registrationFactory.create();
+    mockAuth = authFactory.create();
+
+    token = mockAuth.token;
+    firstName = mockRegistration.customer.firstName;
+    lastName = mockRegistration.customer.lastName;
+    email = mockRegistration.customer.email;
+    password = mockRegistration.password;
+    mockLoginInfo = {email, password};
+  });
+
+  it('should be created', () => {
+    expect(effects).toBeTruthy();
+  });
+
+  describe('when AuthLoginAction is triggered', () => {
```

Comment:
I think we should try to write the describe part of the tests so it's a little easier to discern the purpose of the code. For example, here:
```
describe - login$ - when the user logs in.
   describe - and the login is successful.
       it - should notify state that the login was successful.
   describe - and the login fails.
       it - should notify state that the login failed.
```

Reply:
lgtm otherwise

Reply:
changed